frontend/src/pages/projects/screens/spawner/SpawnerFooter.tsx

```diff
@@ -44,12 +44,38 @@ const SpawnerFooter: React.FC<SpawnerFooterProps> = ({
   canEnablePipelines,
 }) => {
   const [errorMessage, setErrorMessage] = React.useState('');
+
   const {
     dashboardConfig: {
       spec: { notebookController },
     },
+    appContextStorageClasses,
   } = React.useContext(AppContext);
   const tolerationSettings = notebookController?.notebookTolerationSettings;
+
+  const defaultClusterStorageClasses = appContextStorageClasses.filter((storageclass) =>
+    storageclass.metadata.annotations?.['storageclass.kubernetes.io/is-default-class']?.includes(
+      'true',
+    ),
+  );
+
+  const configStorageClassName = notebookController?.storageClassName ?? '';
+
+  let storageClassNameForPVCAssembly = '';
+  if (defaultClusterStorageClasses.length === 0 && configStorageClassName != '') {
+    storageClassNameForPVCAssembly = appContextStorageClasses.filter((storageclass) =>
+      storageclass.metadata.name.includes(configStorageClassName),
+    )[0]?.metadata?.name;
+    if (storageClassNameForPVCAssembly === undefined) {
+      // eslint-disable-next-line no-console
+      console.error(
+        'no cluster default storageclass set and notebooks.storageClassName entry is not in list of cluster StorageClasses',
+      );
+    }
+  } else if (defaultClusterStorageClasses.length != 0) {
+    storageClassNameForPVCAssembly = defaultClusterStorageClasses[0]?.metadata?.name;
+  }
```

Comment:
For code sanity, i think we need to make this into a "custom hook". It'll come with two main benefits.

1. Co-locate all this code in a single spot (we really only want the 1 answer in this function, not the logic to determine it)
2. Being a custom hook, you can pull from the `React.useContext(AppContext);` in it & get your list of storage classes

So I suggest making a `usePreferredStorageClass` hook and place the logic here inside that hook.

Make the hook return a singular value -- either `undefined` or a `StorageClassKind`.

(using the exit-early strat of functions) Logic can be something like:

1. Do we have a default? yes, return `undefined`
2. Do we have a storageClass value in our DashboardConfig? no, return `undefined`
3. Do we have a matching storageClass to the value in the DashboardConfig? no, console.error & return `undefined`
4. Return the Storage Class value you have matching (this allows flexibility of usage, we can determine at this point that we only want the `metadata.name` if we want)

Then feed that into the PVC method with the `metadata.name` (see my next comment)

Reply:
custom hook concept, interesting, yeah

Reply:
ok I have found a good article regarding custom hooks: https://dev.to/sulistef/how-to-create-a-custom-react-hook-to-fetch-an-api-using-typescript-ioi

In odh.dashboard project, where in the folder structure should I place the file containing the hook? Are there already other custom hooks?

Reply:
Place it next to the component is fine -- we are still working through cleanup of things, but you're the only one to use this hook right now so it can just sit next to the SpawnerFooter.tsx file.

Reply:
it might be that in future, Jupyter tile could use the hook, too, but then, you can keep working on the location, and so on. totally ok with placing it near SpawnerFooter.

Reply:
> (using the exit-early strat of functions)

yes, doing so makes the code and determination much less complex, reducing the need for too-deep nesting.